src/creatures/players/cyclopedia/player_title.cpp

```diff
@@ -0,0 +1,270 @@
+	const auto &unlockedTitles = getUnlockedKV()->keys();
+	g_logger().debug("[{}] - Loading unlocked titles: {}", __FUNCTION__, unlockedTitles.size());
+	for (const auto &titleName : unlockedTitles) {
+		const Title &title = g_game().getTitleByName(titleName);
+		if (title.m_id == 0) {
+			g_logger().error("[{}] - Title {} not found.", __FUNCTION__, titleName);
+			continue;
+		}
+
+		auto femaleLog = !title.m_femaleName.empty() ? ("/" + title.m_femaleName) : "";
+		// g_logger().debug("[{}] - Title {}{} found for player {}.", __FUNCTION__, title.m_maleName, femaleLog, m_player.getName());
+
+		m_titlesUnlocked.emplace_back(title, getUnlockedKV()->get(titleName)->getNumber());
+	}
+}
+
+const std::shared_ptr<KV> &PlayerTitle::getUnlockedKV() {
+	if (m_titleUnlockedKV == nullptr) {
+		m_titleUnlockedKV = m_player.kv()->scoped("titles")->scoped("unlocked");
+	}
+
+	return m_titleUnlockedKV;
+}
+
+// Title Calculate Functions
+bool PlayerTitle::checkGold(uint32_t amount) {
+	return m_player.getBankBalance() >= amount;
+}
+
+bool PlayerTitle::checkMount(uint32_t amount) {
+	uint8_t total = 0;
+	for (const auto &mount : g_game().mounts.getMounts()) {
+		if (m_player.hasMount(mount)) {
+			total = total++;
+		}
+	}
+	return total >= amount;
+}
+
+bool PlayerTitle::checkOutfit(uint32_t amount) {
+	return m_player.outfits.size() >= amount;
+}
+
+bool PlayerTitle::checkLevel(uint32_t amount) {
+	return m_player.getLevel() >= amount;
+}
+
+bool PlayerTitle::checkHighscore(uint8_t skill) {
+	// todo cyclopledia
+	return false;
+	//	return g_game().getH() m_player.getLevel() >= amount;
+}
+
+bool PlayerTitle::checkBestiary(uint16_t race) {
+	return m_player.isCreatureUnlockedOnTaskHunting(g_monsters().getMonsterTypeByRaceId(race));
+}
+
+bool PlayerTitle::checkLoginStreak(uint32_t amount) {
+	auto streakKV = m_player.kv()->scoped("daily-reward")->get("streak");
+	return streakKV && streakKV.has_value() && static_cast<uint8_t>(streakKV->getNumber()) >= amount;
+}
+
+bool PlayerTitle::checkTask(uint32_t amount) {
+	return m_player.getTaskHuntingPoints() >= amount;
+}
+
+bool PlayerTitle::checkMap(uint32_t amount) {
+	// todo cyclopledia
+	return false;
+}
+
+bool PlayerTitle::checkQuest(TitleStorage storage) {
+	return !storage.kvKey.empty()
+		? m_player.kv()->get(storage.kvKey)->getNumber() == storage.value
+		: m_player.getStorageValue(storage.key) == storage.value;
+}
+
+bool PlayerTitle::checkOther(std::string name) {
+	if (name == "Admirer of the Crown") {
+		// todo
+		return false;
+	} else if (name == "Big Spender" && m_player.canWear(1211, 3) && m_player.canWear(1210, 3)) {
+		return true;
+	} else if (name == "Guild Leader") {
+		auto rank = m_player.getGuildRank();
+		return rank && rank->level == 3;
+	} else if (name == "Jack of all Taints") {
+		// todo
+		return false;
+	} else if (name == "Reigning Drome Champion") {
+		// todo
+		return false;
+	}
+	return false;
+}
+
+/*int GameFunctions::luaGameGetHighscoresLeaderId(lua_State* L) {
```

Comment:
Do you have a reason to be commented? Maybe it would be interesting to remove the function if it is not used or remove the comment